test/test_suite.py

```diff
@@ -509,3 +550,196 @@ def _literal_round_trip(self, type_, input_, output, filter_=None):
+            ]
+        ).difference(["unique_c_a_b"])
+
+        assert not idx_names.intersection(uq_names)
+        if names_that_duplicate_index:
+            eq_(names_that_duplicate_index, idx_names)
+            eq_(uq_names, set())
+
+    @testing.provide_metadata
+    def test_unique_constraint_raises(self):
+        """
+        Checking that unique constraint creation
+        fails due to a ProgrammingError.
+        """
+        Table(
+            "user_tmp_failure",
+            self.metadata,
+            Column("id", sqlalchemy.INT, primary_key=True),
+            Column("name", sqlalchemy.VARCHAR(50)),
+            sqlalchemy.UniqueConstraint("name", name="user_tmp_uq"),
+        )
+
+        with pytest.raises(spanner_dbapi.exceptions.ProgrammingError):
+            self.metadata.create_all()
+
+    @testing.provide_metadata
+    def _test_get_table_names(self, schema=None, table_type="table", order_by=None):
+        """
+        SPANNER OVERRIDE:
+
+        Spanner doesn't support temporary tables, so real tables are
+        used for testing. As the original test expects only real
+        tables to be read, and in Spanner all the tables are real,
+        expected results override is required.
+        """
+        _ignore_tables = [
+            "comment_test",
+            "noncol_idx_test_pk",
+            "noncol_idx_test_nopk",
+            "local_table",
+            "remote_table",
+            "remote_table_2",
+        ]
+        meta = self.metadata
+
+        insp = inspect(meta.bind)
+
+        if table_type == "view":
+            table_names = insp.get_view_names(schema)
+            table_names.sort()
+            answer = ["email_addresses_v", "users_v"]
+            eq_(sorted(table_names), answer)
+        else:
+            if order_by:
+                tables = [
+                    rec[0]
+                    for rec in insp.get_sorted_table_and_fkc_names(schema)
+                    if rec[0]
+                ]
+            else:
+                tables = insp.get_table_names(schema)
+            table_names = [t for t in tables if t not in _ignore_tables]
+
+            if order_by == "foreign_key":
+                answer = ["users", "user_tmp", "email_addresses", "dingalings"]
+                eq_(table_names, answer)
+            else:
+                answer = ["dingalings", "email_addresses", "user_tmp", "users"]
+                eq_(sorted(table_names), answer)
+
+    @testing.requires.table_reflection
+    def test_numeric_reflection(self):
+        """
+        SPANNER OVERRIDE:
+
+        Spanner defines NUMERIC type with the constant precision=38
+        and scale=9. Overriding the test to check if the NUMERIC
+        column is successfully created and has dimensions
+        correct for Cloud Spanner.
+        """
+        for typ in self._type_round_trip(Numeric(18, 5)):
+            assert isinstance(typ, Numeric)
+            eq_(typ.precision, 38)
+            eq_(typ.scale, 9)
```

Comment:
In this override I'm keeping the original `(18, 5)` dimensions to check that dialect will create NUMERIC without trying to do it like `NUMERIC(18, 5)`, as it'll cause syntax error in Spanner.